.github/workflows/deploy.yml

```
@@ -51,17 +70,27 @@ jobs:
   deploy:
     runs-on: ubuntu-latest
     needs: build
-
-    permissions:
-      pages: write
-      id-token: write
-
+    
     environment:
-      name: github-pages
-      url: ${{ steps.deployment.outputs.page_url }}
+      name: ${{ github.ref == 'refs/heads/main' && 'github-pages' || 'github-pages-preview' }}
+      url: ${{ github.ref == 'refs/heads/main' && steps.deployment.outputs.page_url || format(steps.deployment.outputs.page_url, '/preview', github.event.pull_request.number) }}
+    
+    outputs:
+      deployment_url: ${{ env.url }}

```

```suggestion
      deployment_url: ${{ steps.deployment.outputs.page_url }}
```

You don't set `env.url` anywhere. Note `env` and `environment` are unrelated.